app/services/badge/bestowment_service.rb

```diff
@@ -0,0 +1,55 @@
+# Handles creating and updating bestowments for a (badge, user) pair.
+class Badge
+  class BestowmentService
+    attr_reader :badge, :user
+
+    # @param [Class<Badge>] the badge to handle bestowments for
+    # @param [User] the user to bestow this badge upon
+    def initialize(badge, user)
+      @badge = badge
+      @user = user
+    end
+
+    # Actually creates/updates Bestowments for the badge
+    def run!
+      tracked_ranks.each do |(rank, progress)|
+        bestowment = bestowment_for(rank).first_or_initialize
+        # Ratcheting - progress can only increase, never decrease.
+        bestowment.progress = [bestowment.progres, progress].max
```

Comment:
So If someone deletes  their comment,post or likes, they will maintain their progress for a badge?